Inferences/ForwardDemodulation.cpp

```diff
@@ -167,9 +163,23 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           }
         }
   #endif
-        if(!preordered && (_preorderedOnly || ordering.compare(trm,rhsS)!=Ordering::GREATER) ) {
+
+        auto subs = qr.unifier;
+        ASS(subs->isIdentityOnQueryWhenResultBound());
+
+        Applicator<true> varSubst(subs.ptr(), subst);
```

Comment:
Could we maybe give `subs` and `subst` different names indicating their use cases better? 
Like `eqSortSubs` and `unifier` or something similar?